File: DheerajKumar/problem33.cpp
//33. WAP to take input a number from the user and check whether it is palindrome.
#include<iostream>
using namespace std;

int reverse(int number){
    if(number <= 0){
        return -1;
    }
    
    int lastDigit = 0, reverseNumber = 0;
    while(number > 0){
        lastDigit = number % 10;
        reverseNumber = (reverseNumber * 10) + lastDigit;
        number = number/10; 
    }

    return reverseNumber;
}

int main(){
    int number;
    cout<<"Enter the number -> ";
    cin>> number;

    if(number == reverse(number)){
        cout<<"The number is palindrome";
    }else{
        cout<<"The number is  not a palindrome";
    }
}

File: LovelyKumari/BasicProgramming/Problem33.java
import java.util.Scanner;

public class Problem33 {
    //WAP to take input a number from the user and check whether it is palindrome.
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter a number: ");
        int num = scanner.nextInt();

        int reverse = 0 ;
        int original=num;
        while (num != 0) {
            int digit = num % 10;     
            reverse = reverse * 10 + digit;  
            num /= 10;  
        }
        if(original==reverse){
            System.out.println(original+" is a palindrome number.");
        }else{
            System.out.println(original+" is not a palindrome number.");
        }
    }
    
}
